### PR TITLE
1.17.0 release

### PR DESCRIPTION
**Pull request by mattklein123**

### CODE REVIEW

**Comment by mattklein123** [comment]
cc @envoyproxy/maintainers to have a look through the release notes.

**Review by jmarantz — DISMISSED** [review]

File: docs/root/version_history/current.rst [hunk 1]
```diff
@@ -13,16 +13,13 @@ Minor Behavior Changes
...
 * decompressor: set the default value of window_bits of the decompressor to 15 to be able to decompress responses compressed by a compressor with any window size.
 * expr filter: added `connection.termination_details` property support.
-* ext_authz filter: disable `envoy.reloadable_features.ext_authz_measure_timeout_on_check_created` by default.
 * ext_authz filter: the deprecated field :ref:`use_alpha <envoy_api_field_config.filter.http.ext_authz.v2.ExtAuthz.use_alpha>` is no longer supported and cannot be set anymore.
 * formatter: the :ref:`text_format <envoy_v3_api_field_config.core.v3.SubstitutionFormatString.text_format>` field no longer requires at least one byte, and may now be the empty string. It has also become deprecated: see Deprecated section.
 * grpc_web filter: if a `grpc-accept-encoding` header is present it's passed as-is to the upstream and if it isn't `grpc-accept-encoding:identity` is sent instead. The header was always overwriten with `grpc-accept-encoding:identity,deflate,gzip` before.
```
Comment (alyssawilk) [inline docs/root/version_history/current.rst:20]: aside: should this have been guarded given it's a data plane change?  Our policy only says most data plane changes so maybe it was small enough it was intentional? cc @lizan

File: docs/root/version_history/current.rst [hunk 2]
```diff
@@ -13,16 +13,13 @@ Minor Behavior Changes
...
 * http: upstream protocol will now only be logged if an upstream stream was established.
 * jwt_authn filter: added support of Jwt time constraint verification with a clock skew (default to 60 seconds) and added a filter config field :ref:`clock_skew_seconds <envoy_v3_api_field_extensions.filters.http.jwt_authn.v3.JwtProvider.clock_skew_seconds>` to configure it.
```
Comment (alyssawilk) [inline docs/root/version_history/current.rst:22]: Jwt -> JWT?

File: docs/root/version_history/current.rst [hunk 3]
```diff
@@ -102,9 +98,9 @@ New Features
...
-* start_tls: :ref:`transport socket<envoy_v3_api_msg_extensions.transport_sockets.starttls.v3.StartTlsConfig>` which starts in clear-text but may programatically be converted to use tls.
+* start_tls: add new :ref:`transport socket<envoy_v3_api_msg_extensions.transport_sockets.starttls.v3.StartTlsConfig>` which starts in clear-text but may programatically be converted to use tls.
 * tcp: added a new :ref:`envoy.overload_actions.reject_incoming_connections <config_overload_manager_overload_actions>` action to reject incoming TCP connections.
-* thrift_proxy: added a new :ref: `payload_passthrough <envoy_v3_api_field_extensions.filters.network.thrift_proxy.v3.ThriftProxy.payload_passthrough>` option to skip decoding body in the Thrift message.
+* thrift_proxy: added a new :ref:`payload_passthrough <envoy_v3_api_field_extensions.filters.network.thrift_proxy.v3.ThriftProxy.payload_passthrough>` option to skip decoding body in the Thrift message.
 * tls: added support for RSA certificates with 4096-bit keys in FIPS mode.
 * tracing: added SkyWalking tracer.
```
Comment (alyssawilk) [inline docs/root/version_history/current.rst:105]: config link? (cc @lizan)

File: docs/root/version_history/current.rst [hunk 4]
```diff
@@ -102,9 +98,9 @@ New Features
...
 * tracing: added support for setting the hostname used when sending spans to a Zipkin collector using the :ref:`collector_hostname <envoy_v3_api_field_config.trace.v3.ZipkinConfig.collector_hostname>` field.
```
Comment (alyssawilk) [inline docs/root/version_history/current.rst:106]: for gzip filter deprecation should we link to the replacement, or assume they can find from prior release notes?

**Review by alyssawilk — COMMENTED** [review]
Looks great!  handful of mostly optional comments below.

File: docs/root/version_history/current.rst [hunk 5]
```diff
@@ -13,16 +13,13 @@ Minor Behavior Changes
...
 * ext_authz filter: the deprecated field :ref:`use_alpha <envoy_api_field_config.filter.http.ext_authz.v2.ExtAuthz.use_alpha>` is no longer supported and cannot be set anymore.
 * formatter: the :ref:`text_format <envoy_v3_api_field_config.core.v3.SubstitutionFormatString.text_format>` field no longer requires at least one byte, and may now be the empty string. It has also become deprecated: see Deprecated section.
 * grpc_web filter: if a `grpc-accept-encoding` header is present it's passed as-is to the upstream and if it isn't `grpc-accept-encoding:identity` is sent instead. The header was always overwriten with `grpc-accept-encoding:identity,deflate,gzip` before.
 * http: upstream protocol will now only be logged if an upstream stream was established.
 * jwt_authn filter: added support of Jwt time constraint verification with a clock skew (default to 60 seconds) and added a filter config field :ref:`clock_skew_seconds <envoy_v3_api_field_extensions.filters.http.jwt_authn.v3.JwtProvider.clock_skew_seconds>` to configure it.
-* kill_request: enable a way to configure kill header name in KillRequest proto.
```
[the remark this answers is not in the document]
Comment (mattklein123) [reply docs/root/version_history/current.rst:25]: This was added in the same release, it's linked below, so I don't think we need to call this out also.

**Review by alyssawilk — APPROVED** [review]

**Pull request merged**